### PR TITLE
PTX-12071 Implement method to determine Operator and PX versions

### PR DESCRIPTION
**Pull request by nikolaypopov**
Implement method to determine Operator and PX versions for future new CCM testing and upgrades

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -2335,6 +2342,96 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
...
+	var image string
+
+	NamespaceList, err := coreops.Instance().ListNamespaces(labelSelector)
+	if err != nil {
+		return image, err
+	}
+
+	for _, ns := range NamespaceList.Items {
+		operatorDeployment, err := appops.Instance().GetDeployment("portworx-operator", ns.Name)
+		if errors.IsNotFound(err) {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:2419]: return if hit other errors

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -2335,6 +2342,96 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
...
+	image, err := getPxOperatorImage()
+	if err != nil {
+		return nil, err
+	}
+
+	// We may run the automation on operator installed using private images,
+	// so assume we are testing the latest operator version if failed to parse the tag
+	var versionTag string
+	versionTag = strings.Split(image, ":")[1]
+	if strings.Contains(versionTag, "-dev") {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:2394]: with `-dev` we could still parse the version, e.g. version 1.9.0-dev, and can be used to compare versions. so we could skip this step.

File: pkg/util/test/util.go [hunk 3]
```diff
@@ -2335,6 +2342,96 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
...
+func GetPxOperatorVersion() (*version.Version, error) {
+	image, err := getPxOperatorImage()
+	if err != nil {
+		return nil, err
+	}
+
+	// We may run the automation on operator installed using private images,
+	// so assume we are testing the latest operator version if failed to parse the tag
+	var versionTag string
+	versionTag = strings.Split(image, ":")[1]
```
Comment (pure-jliao) [inline pkg/util/test/util.go:2393]: nit: use last one instead. but should be fine without custom registry.

File: pkg/util/test/util.go [hunk 4]
```diff
@@ -2335,6 +2342,96 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
 	return ValidateTelemetryUninstalled(pxImageList, cluster, timeout, interval)
 }
 
+// GetPortworxVersion returns Portworx version from StorageCluster
```
[the remark this answers is not in the document]
Comment (nikolaypopov) [reply pkg/util/test/util.go:2345]: Good point, should have looked and saved time and yeah there are import issues so I will copy this

**Review by pure-jliao — APPROVED** [review]

**Pull request merged**